alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexMonitorAction.kt

```diff
@@ -494,6 +504,27 @@ class TransportIndexMonitorAction @Inject constructor(
             if (request.monitor.enabled && currentMonitor.enabled)
                 request.monitor = request.monitor.copy(enabledTime = currentMonitor.enabledTime)
 
+            // On update monitor check which backend roles to associate to the monitor
+            if (user != null) {
```

Comment:
Could you add tests for these scenarios?

Reply:
yes